tests/common/accuracy_control/test_algorithm.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from dataclasses import dataclass
+from unittest.mock import Mock
+
+import numpy as np
+import pytest
+
+from nncf.common.graph.graph import NNCFNode
+from nncf.quantization.algorithms.accuracy_control.algorithm import QuantizationAccuracyRestorer
+from nncf.quantization.algorithms.accuracy_control.algorithm import QuantizationAccuracyRestorerReport
+from nncf.quantization.algorithms.accuracy_control.algorithm import _create_message
+from tests.common.accuracy_control.backend import AABackendForTests
+from tests.common.quantization.mock_graphs import get_mock_model_graph_with_mergeable_pattern
+
+
+def test_create_message():
+    nodes = [
+        NNCFNode(
+            {
+                NNCFNode.NODE_NAME_ATTR: "node_name_1",
+            }
+        ),
+        NNCFNode(
+            {
+                NNCFNode.NODE_NAME_ATTR: "node_name_2",
+            }
+        ),
+    ]
+    result = _create_message(nodes)
+    assert isinstance(result, str)
+    assert result == "\n".join(["\t" + node.node_name for node in nodes])
+
+
+@pytest.fixture
+def setup_quantization_accuracy_restorer_report():
+    report = QuantizationAccuracyRestorerReport()
+    node1 = NNCFNode(
+        {
+            NNCFNode.NODE_NAME_ATTR: "node_name_1",
+        }
+    )
+    node2 = NNCFNode(
+        {
+            NNCFNode.NODE_NAME_ATTR: "node_name_2",
+        }
+    )
+    report.removed_groups = [type("MockGroup", (), {"quantizers": [node1], "operations": [node2]})]
+    return report
+
+
+def test_removed_quantizers(setup_quantization_accuracy_restorer_report):
+    assert setup_quantization_accuracy_restorer_report.removed_quantizers[0].node_name == "node_name_1"
+
+
+def test_reverted_operations(setup_quantization_accuracy_restorer_report):
+    assert setup_quantization_accuracy_restorer_report.reverted_operations[0].node_name == "node_name_2"
```

Comment:
I suggest representing the test in the following way:

```python
@pytest.mark.parametrize(
    # test data here
)
def test_quantization_accuracy_restorer_report_properties(
    removed_groups,
    expected_removed_quantizers,
    expected_reverted_operations
):
    report = QuantizationAccuracyRestorerReport()
    report.removed_groups = removed_groups
    assert report.removed_quantizers == expected_removed_quantizers
    assert report.expected_reverted_operations == expected_reverted_operations
    
```
It allows us to quickly understand what the input data for the test is and what the expected values are.